python_modules/dagster/dagster/_core/definitions/reconstruct.py

```diff
@@ -434,29 +434,48 @@ def make_bar_job():
             )
         )
 
+    code_pointer = _find_code_pointer(target)
+    if code_pointer:
+        return bootstrap_standalone_recon_pipeline(code_pointer)
+    else:
+        raise DagsterInvariantViolationError(
+            "reconstructable() can not reconstruct jobs or pipelines defined in interactive "
+            "environments like <stdin>, IPython, or Jupyter notebooks. "
+            "Use a pipeline defined in a module or file instead, or use build_reconstructable_job."
+        )
+
+
+def reconstructable_repository(target: "RepositoryDefinition") -> ReconstructableRepository:
+    code_pointer = _find_code_pointer(target)
+
+    if code_pointer:
+        return ReconstructableRepository(code_pointer)
+    else:
+        raise DagsterInvariantViolationError(
+            "reconstructable_repository() can only reconstruct repositories that are defined "
+            "within a module, at module scope. It cannot not reconstruct repositories defined in "
+            "interactive environments like <stdin>, IPython, or Jupyter notebooks."
```

Comment:
Given that the entrypoint for this error is [1], I don't think it makes sense to make the error in terms of `reconstructable_repository`, instead keeping things focused on the fact that the repo was not constructed at module scope.

Reply:
👍